server/controllers/events.js

```diff
@@ -84,6 +84,24 @@ const getUsersForEvent = async (req, res) => {
   }
 }
 
+const getEventsForUser = async (req, res) => {
+  let user_id = req.params.id;
+  try {
+    await models.getEventsForUser(user_id).then(async (data) => {
+      let events = [];
+      for (let i = 0; i < data.rows.length; i++) {
+        await models.getSpecificEvent(data.rows[i].events_id)
+        .then((data) => { events.push(data.rows[0]); });
```

Comment:
Since `await` turns asynchronous code into synchronous code, you can just assign the result to a variable instead of using `.then`.

```
const data = await models.getSpecificEvent(data.rows[i].events_id);
events.push(data.rows[0]);
res.send(data);
```

Reply:
Implemented this, but had to re-declare the variable inside the loop to get it to work.